GitHub_LearningTests/Feature/StartUp/Application/StartUpControllerTests.swift

```diff
@@ -0,0 +1,226 @@
+        let credentialKeychainController = CredentialKeychainControllerSuccessStub()
+        let dashboardLauncherController = DashboardLauncherControllerDummy()
+        setUpTests(
+            loadOnboardingFinishedFacade: facade,
+            credentialKeychainController: credentialKeychainController,
+            dashboardLauncherController: dashboardLauncherController
+        )
+        
+        // when
+        sut.startUp()
+        
+        //Then
+        XCTAssertNotNil(dashboardLauncherController.launched)
+        XCTAssert(dashboardLauncherController.launched == true)
+        XCTAssert(dashboardLauncherController.email == credentialKeychainController.loadedCredentials)
+        XCTAssert(dashboardLauncherController.point == DashboardLaunchPoint.startUp)
+    }
+    
+    func test_givenLoadCredentailsIsCalledWithFailure_whenStartUpIsCalled_thenLoginLauncherCallsLaunchLoginAfter() {
+        // given
+        let facade = LoadOnboardingFinishedFacadeSuccessTrueStub()
+        let credentialKeychainController = CredentialKeychainControllerFailureStub()
+        let loginLauncherController = LoginLauncherControllerDummy()
+        setUpTests(
+            loadOnboardingFinishedFacade: facade,
+            credentialKeychainController: credentialKeychainController,
+            loginLauncherController: loginLauncherController
+        )
+        
+        // when
+        sut.startUp()
+        
+        //Then
+        XCTAssert(credentialKeychainController.isLoaded == false)
+        XCTAssertNotNil(loginLauncherController.launched)
+        XCTAssert(loginLauncherController.launched == true)
+        XCTAssert(loginLauncherController.point == LoginLaunchPoint.start)
+    }
+    
+    private class LoadOnboardingFinishedFacadeDummy: LoadOnboardingFinishedFacade {
+        func load(_ request: LoadOnboardingFinishedRequest, completion: @escaping ((Result<LoadOnboardingFinishedResponse>) -> Void)) {}
+    }
+    
+    private class LoadOnboardingFinishedFacadeSuccessTrueStub: LoadOnboardingFinishedFacade {
+        func load(_ request: LoadOnboardingFinishedRequest, completion: @escaping ((Result<LoadOnboardingFinishedResponse>) -> Void)) {
+            completion(.success(response: LoadOnboardingFinishedResponse(isFinished: true)))
+        }
+    }
+    
+    private class LoadOnboardingFinishedFacadeSuccessFalseStub: LoadOnboardingFinishedFacade {
+        func load(_ request: LoadOnboardingFinishedRequest, completion: @escaping ((Result<LoadOnboardingFinishedResponse>) -> Void)) {
+            completion(.success(response: LoadOnboardingFinishedResponse(isFinished: false)))
+        }
+    }
+    
+    private class LoadOnboardingFinishedFacadeFailureStub: LoadOnboardingFinishedFacade {
+        func load(_ request: LoadOnboardingFinishedRequest, completion: @escaping ((Result<LoadOnboardingFinishedResponse>) -> Void)) {
+            completion(.failure(error: LocalStorageError.loadFailed("Load Failed")))
+        }
+    }
```

Comment:
To stejne. Pokud bys v initu predal "navratovou hodnotu" ty facady, tak si usetris 2 fake tridy.